src/CondenserDotNet.Server/CurrentState.cs

```diff
@@ -2,91 +2,78 @@
-                    returnValue.Http200Responses += stat.Http200Responses;
-                    returnValue.Http300Responses += stat.Http300Responses;
-                    returnValue.Http400Responses += stat.Http400Responses;
-                    returnValue.Http500Responses += stat.Http500Responses;
-                }
-            }
-            catch (Exception ex)
+            var stats = new StatsSummary();
+
+            lock (_lock)
             {
-                _logger?.LogWarning(0, ex, "Exception while calculating the health summary");
+                UpTime = DateTime.UtcNow - _startedTime;
+
+                stats.Http100Responses = Http100Responses;
+                stats.Http200Responses = Http200Responses;
+                stats.Http300Responses = Http300Responses;
+                stats.Http400Responses = Http400Responses;
+                stats.Http500Responses = Http500Responses;
+                stats.HttpUnknownResponse = HttpUnknownResponse;
+                stats.UpTime = UpTime;
             }
-            returnValue.UpTime = DateTime.UtcNow - _startedTime;
-            return returnValue;
-        }
 
-        public struct Summary
-        {
-            public int Http100Responses;
-            public int Http200Responses;
-            public int Http300Responses;
-            public int Http400Responses;
-            public int Http500Responses;
-            public TimeSpan UpTime;
-            public int HttpUnknownResponse;
+            return stats;
         }
 
+        public int Http100Responses { get; private set; }
+        public int Http200Responses { get; private set; }
+        public int Http300Responses { get; private set; }
+        public int Http400Responses { get; private set; }
+        public int Http500Responses { get; private set; }
+        public TimeSpan UpTime { get; private set; }
+        public int HttpUnknownResponse { get; private set; }
+
+        private readonly object _lock = new object();
+
         public void RecordResponse(int responseCode)
         {
-            responseCode = responseCode / 100;
-            responseCode = responseCode * 100;
-            switch (responseCode)
+            lock (_lock)
             {
-                case 500:
-                    Stats.Http500Responses++;
-                    break;
-                case 400:
-                    Stats.Http400Responses++;
-                    break;
-                case 300:
-                    Stats.Http300Responses++;
-                    break;
-                case 200:
-                    Stats.Http200Responses++;
-                    break;
-                case 100:
-                    Stats.Http100Responses++;
-                    break;
-                default:
-                    Stats.HttpUnknownResponse++;
-                    break;
+                responseCode = responseCode / 100;
+                responseCode = responseCode * 100;
```

Comment:
Could avoid the multiplication by changing the case to 5,4,3,2,1 seeing as it will be a hotpath